src/pages/Login/LoginState.js

```diff
@@ -0,0 +1,265 @@
+			setNeedPrfRetry(false);
+			setPrfRetryAccepted(true);
+			setResolvePrfRetryPrompt(null);
+		});
+	};
+
+	const onLogin = useCallback(
+		async (cachedUser) => {
+			const result = await api.loginWebauthn(keystore, promptForPrfRetry, cachedUser);
+			if (result.ok) {
+
+				navigate(from, { replace: true });
+
+			} else {
+				// Using a switch here so the t() argument can be a literal, to ease searching
+				switch (result.val) {
+					case 'loginKeystoreFailed':
+						setError(t('loginSignup.loginKeystoreFailed'));
+						break;
+
+					case 'passkeyInvalid':
+						setError(t('loginSignup.passkeyInvalid'));
+						break;
+
+					case 'passkeyLoginFailedTryAgain':
+						setError(t('loginSignup.passkeyLoginFailedTryAgain'));
+						break;
+
+					case 'passkeyLoginFailedServerError':
+						setError(t('loginSignup.passkeyLoginFailedServerError'));
+						break;
+
+					default:
+						throw result;
+				}
+			}
+		},
+		[api, keystore, navigate, t],
+	);
+
+	const onSubmit = async (event) => {
+		event.preventDefault();
+
+		setError();
+		setInProgress(true);
+		setIsSubmitting(true);
+
+		await onLogin();
+
+		setInProgress(false);
+		setIsSubmitting(false);
+	};
+
+	const onLoginCachedUser = async (cachedUser) => {
+		setError();
+		setInProgress(true);
+		setIsSubmitting(true);
+		await onLogin(cachedUser);
+		setInProgress(false);
+		setIsSubmitting(false);
+	};
+
+	return (
+		<form onSubmit={onSubmit}>
+			<>
+				<ul className=" p-2">
+					{filteredUser && (
+						<div className='flex flex-row gap-4 justify-center mr-2'>
+							<GetButton
+								content={
+									<>
+										{isSubmitting ? t('loginSignup.submitting') : t('common.cancel')}
+									</>
+								}
+								onClick={() => navigate('/')}
+								variant="cancel"
+								disabled={isSubmitting}
+								additionalClassName='w-full'
+							/>
+							<GetButton
+								content={
+									<>
+										<GoPasskeyFill className="inline text-xl mr-2" />
+										{isSubmitting ? t('loginSignup.submitting') : t('common.continue')}
+									</>
+								}
+								onClick={() => onLoginCachedUser(filteredUser)}
+								variant="primary"
+								disabled={isSubmitting}
+								additionalClassName='w-full'
+							/>
+						</div>
+					)}
+				</ul>
+				{error && <div className="text-red-500 pt-4">{error}</div>}
+			</>
+		</form>
+	);
+};
+
+const loginState = () => {
+	const { isOnline } = useContext(OnlineStatusContext);
+	const api = useApi(isOnline);
+	const { t } = useTranslation();
+	const location = useLocation();
+
+	const [isSubmitting, setIsSubmitting] = useState(false);
+	const [isContentVisible, setIsContentVisible] = useState(false);
+	const [filteredUser, setFilteredUser] = useState(null);
+	const navigate = useNavigate();
+	const keystore = useLocalStorageKeystore();
+	const cachedUsers = keystore.getCachedUsers();
+	const from = location.state?.from;
+
+	useEffect(() => {
+		const queryParams = new URLSearchParams(from.search);
```

Comment:
Loading `/login-state?state=...` directly produces this error:

```
TypeError: Cannot read properties of undefined (reading 'search')
    at LoginState.js:162:1
```

Hm, I see that the `useEffect` lists `location.search` as a dependency, but then reads `from.search` instead of `location.search`. Which one is the intent?